src/corelib/mimetypes/qmimeprovider.cpp

```diff
@@ -647,7 +647,7 @@ QMimeXMLProvider::QMimeXMLProvider(QMimeDatabasePrivate *db, InternalDatabaseEnu
 #elif defined(MIME_DATABASE_IS_GZIP)
     std::unique_ptr<char []> uncompressed(new char[size]);
     z_stream zs = {};
-    zs.next_in = mimetype_database;
+    zs.next_in = const_cast<Bytef *>(mimetype_database);
```

Comment:
While it should work, I'm wondering whether it's better to avoid casting here.

Something like:

QString mimetypeDatabase{mimetype_database};
zs.next_in = mimetypeDatabase.data();